htdocs/content/crypto.php

```diff
@@ -193,75 +193,85 @@
 										// Do the scan if chance >= 0
 										if ($chance >= 0)
 										{
-											$decryptlevel = ($cryptoCenterLevel-$op_jam) + (0.75*($self_spy + $self_computer - $op_stealth - $op_computer)) + mt_rand(0,2)-1;
-
-											// Decrypt level
-											// < 0 Only show that there are some fleets
-											// 0 <= 10 Show that there are x fleets
-											// 10 <= 15 Show that there are x fleets from y belonging to z, show hour
-											// 15 <= 20 Also show ship types, show mninutes with +- 15 mins
-											// 20 <= 25 Also show count of ships and time in minutes
-											// 25 <= 30 Also show count of every ship and exact time
-											// >30 Show action
+											$decryptlevel = ($cryptoCenterLevel-$op_jam) +
+												(0.75*($self_spy + $self_computer - $op_stealth - $op_computer)) +
+												mt_rand($cfg->value("crypto_rand_mod_min"),$cfg->value("crypto_rand_mod_max"));
 
 											$out="[b]Flottenscan vom Planeten ".$target->name()."[/b] (".$sx."/".$sy." : ".$cx."/".$cy." : ".$pp.")\n\n";
 
-											$out.="[b]Eintreffende Flotten[/b]\n\n";
-											$fres = dbquery("
-															SELECT
-																id
-															FROM
-																fleet
-															WHERE
-																entity_to=".$target->id()."");
-											if (mysql_num_rows($fres)>0)
+											if ($decryptlevel>=$cfg->value("crypto_number_of_fleets_level"))
 											{
-												if ($decryptlevel<0)
-												{
-													$out.="Es sind Flotten unterwegs\n";
-												}
-												else if ($decryptlevel<10)
-												{
-													$out.="Es sind ".mysql_num_rows($fres)." Flotten unterwegs\n";
-												}
-												else
+												$fres = dbquery("
+												SELECT
+													id
+												FROM
+													fleet
+												WHERE
+													entity_from=".$target->id()."
+													OR entity_to=".$target->id()."");
+												$out.="Es sind ".mysql_num_rows($fres)." Flotten unterwegs.\n\n";
+											}
+											else
+											{
+												$out.="Es sind Flotten unterwegs.\n\n";
```

Comment:
This now shows `Es sind Flotten unterwegs.` if the `$decryptlevel` even if there are no fleets which is wrong information